signac/contrib/schema.py

```diff
@@ -36,70 +39,68 @@ def _collect_by_type(values):
         A mapping of types to a set of input values of that type.
 
     """
-    values_by_type = ddict(set)
+    values_by_type = defaultdict(set)
     for v in values:
         values_by_type[type(v)].add(v)
     return values_by_type
 
 
+def _strip_prefix(key):
+    return key[len("sp.") :]
+
+
+def _remove_dict_placeholder(x):
```

Comment:
I don't think this should be defined here. It belongs with _Index (or Collection until that class is removed).

Reply:
Good idea! I did that. For now I moved to `_index.py` since `collection.py` will eventually be deleted.